ijkmedia/ijkplayer/ff_ffplay.c

```diff
@@ -637,6 +680,27 @@ static void video_image_display2(FFPlayer *ffp)
         ffp->stat.latest_seek_load_duration = (av_gettime() - is->latest_seek_load_start_at) / 1000;
 
     if (vp->bmp) {
+        if (is->subtitle_st) {
+            if (frame_queue_nb_remaining(&is->subpq) > 0) {
+                sp = frame_queue_peek(&is->subpq);
+
+                if (vp->pts >= sp->pts + ((float) sp->sub.start_display_time / 1000)) {
+                    if (!sp->uploaded) {
+                        if (sp->sub.num_rects > 0) {
+                            char buffered_text[4096];
+                            if (sp->sub.rects[0]->text) {
+                                strncpy(buffered_text, sp->sub.rects[0]->text, 4096);
+                            }
+                            else if (sp->sub.rects[0]->ass) {
+                                parse_ass_subtitle(sp->sub.rects[0]->ass, buffered_text);
+                            }
+                            ffp_notify_msg4(ffp, FFP_MSG_TIMED_TEXT, 0, 0, buffered_text);
```

Comment:
I will do some tests on a new callback mechanism to ensure that there're no memory leaks and no life cycle confusion and then post a new commit.